retail-managment-system/rms-ui/src/app/pages/transaction/transaction.component.ts

```diff
@@ -0,0 +1,163 @@
+import { Component, OnInit } from '@angular/core';
+import {TransactionRepository} from 'src/app/domain/transaction/transaction.repository'
+import { Transaction, TransactionType } from 'src/app/domain/transaction/models/transaction'
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Investors } from '../../domain/investors/models/investor'
+import { InvestorsRepository } from 'src/app/domain/investors/investor.repository';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { MatDialog } from '@angular/material/dialog';
+import { ConfirmDialogComponent } from '../../shared/components/confirm-dialog/confirm-dialog.component';
+import { TranslateService } from '@ngx-translate/core';
+
+
+@Component({
+  selector: 'app-transaction',
+  templateUrl: './transaction.component.html',
+  styleUrls: ['./transaction.component.scss']
+})
+export class TransactionComponent implements OnInit {
+
+  transactionForm!: FormGroup
+  allTransactions : Transaction[] = []
+  investorNames : Investors[]=[]
+  transactionTypeEnum = Object.values(TransactionType)
+  displayedColumns: string[] = ['id','investorName','type','amount','date','edit-icon' ,'delete-icon'];
+  currentData!: Transaction
+  isButtonVisible: boolean = true
+  submit: boolean = false;
+  addMesaage!: string
+  updateMessage!:string
+  deleteMessage!:string
+  close!:string
+
+  constructor(
+    private investorsRepository : InvestorsRepository,
+    private transactionRepository : TransactionRepository,
+    private fb :FormBuilder,
+    private SnackBar: MatSnackBar,
+    private dialog: MatDialog,
+    private translate : TranslateService
+    ) {
+    
+  }
+
+  ngOnInit(): void {
+    this.getAllTransactions()
+    this.getInvestorData()
+    this.TransactionForm()
+    this.translate.get([
+      'transactions.add-message',
+      'transactions.update-message',
+      'transactions.delete-message',
+      'transactions.close'])
+      .subscribe(translations =>{
+        this.addMesaage = translations['transactions.add-message']
+        this.updateMessage = translations['transactions.update-message']
+        this.deleteMessage = translations['transactions.delete-message']
+        this.close = translations['transactions.close']
```

Comment:
you can use translate in place you need 
 without return that in property and use it